### PR TITLE
Make Strings Safe Again!

### PR DESCRIPTION
**Pull request by AndreasWebdev**
First version of safeStrings so ffmpeg does not get confused by our inputs!

This references #68

### CODE REVIEW

**Comment by AndreasWebdev** [comment]
Please actually check if the code works, I am currently at work and I don't have much access to other servers (stupid proxy) so I was only able to check if there are any fatal errors before starting the stream (because I also cannot stream to rtmp from here

File: src/stream/safeStrings.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  // ' will be \'
+  safeString.replace(/\'/g, '\\\'');
+
+  // Return safe string
+  return safeString;
+}
+
+module.exports = {
+  forInput: safeStringInput,
+  forIilter: safeStringFilter
```
Comment (torch2424) [inline src/stream/safeStrings.js]: One last bug 😄

Comment (torch2424) [reply src/stream/safeStrings.js]: "forIilter" -> "forFilter"

**Comment by torch2424** [comment]
@AndreasWebdev Yeah I can totally test it! And haha! I was totally replying to you earlier today (Or yesterday for you?) while I was at work haha!

One last thing I found, and then I'll pull down your changes. I would totally fix the typo If I could commit to your branch 😮

**Comment by torch2424** [comment]
If you want, I can totally just add you as a contributor to the repo after this?

File: src/stream/safeStrings.js (hunk 1 above)
Comment (AndreasWebdev) [reply src/stream/safeStrings.js]: am I stupid? sorry for that!

**Comment by AndreasWebdev** [comment]
That'd be great! In your profile it says you're UXE at Google so I am guessing you are on GMT-7 or something (I always notice you're pushing thing's when I am back home or sleeping, I'm on GMT+2) so if we are both online one of us has to be at work :laughing:

**Comment by torch2424** [comment]
Awesome, added you as a collaborator 😄 

And yes I am! Just joined about 2 months ago! But yeah I guess it ends up that way haha! Since I work on an open source project, I'm already always on github so it makes replying easy

File: src/stream/overlayText.js [hunk 2]
```diff
@@ -34,8 +36,9 @@ const getOverlayTextString = async (path, config, typeKey, metadata) => {
     // Check if we have an artist option
     if (overlayConfigObject.artist && overlayConfigObject.artist.enabled) {
       const itemObject = overlayConfigObject.artist;
+      const safeText = safeStrings.forFilter( itemObject.label .  metadata.common.artist);
```
Comment (torch2424) [inline src/stream/overlayText.js]: So I tested this for you 😄 

And I got the following error:

screen shot 2018-10-09 at 1 48 09 am

I noticed you did `const safeText = safeStrings.forFilter(itemObject.label . metadata.common.artist);`

I think that is PHP syntax? 😂 

Thanks again for being so responsive!

File: src/stream/overlayText.js [hunk 3]
```diff
@@ -49,8 +52,9 @@ const getOverlayTextString = async (path, config, typeKey, metadata) => {
     // Check if we have an album option
     if (overlayConfigObject.album && overlayConfigObject.album.enabled) {
       const itemObject = overlayConfigObject.album;
+      const safeText = safeStrings.forFilter( itemObject.label .  metadata.common.album);
```
Comment (torch2424) [inline src/stream/overlayText.js]: See above 😄

File: src/stream/overlayText.js [hunk 4]
```diff
@@ -64,8 +68,9 @@ const getOverlayTextString = async (path, config, typeKey, metadata) => {
     // Check if we have an artist option
     if (overlayConfigObject.song && overlayConfigObject.song.enabled) {
       const itemObject = overlayConfigObject.song;
+      const safeText = safeStrings.forFilter( itemObject.label .  metadata.common.title);
```
Comment (torch2424) [inline src/stream/overlayText.js]: See above 😄

File: src/stream/overlayText.js (hunk 2 above)
Comment (AndreasWebdev) [reply src/stream/overlayText.js]: You're absolutely right! I was working on a PHP project before and JS syntax is so similar I didn't even notice!

**Comment by torch2424** [comment]
Sorry, fell asleep last night with my laptop haha! Will look at this later today, I just woke up :)

**Comment by torch2424** [comment]
Tested and it works! Merging... 😄 

screen shot 2018-10-09 at 11 12 23 pm

**Pull request merged**